Dockerfile

```diff
@@ -20,5 +20,7 @@ RUN python -m poetry install --no-interaction --no-ansi --no-dev \
     # Cleaning poetry installation's cache for production:
     && rm -rf "$POETRY_CACHE_DIR" \
     && pip uninstall -yq poetry
+RUN pip install pycontw-report-generator
```

Comment:
Good to me, and I want to collection more opinions if possible before I move forward with this suggestion. I hope the discussion is not too bike-shedding:

This is essentially two different perspectives:
1. use poetry to manage the basic airflow installation, and we build the other stuff on top of it.
2. use poetry to manage all python packages for the whole airflow docker image

The original docker file removes poetry at the end of the docker file, and this seems to imply "we only want to install a vanilla airflow as a foundation or core service". I am not sure what the motivation is to remove poetry (for smaller docker images?). If we want to choose the option2 (use poetry to manage all packages), I would suggest to add more comments in the Dockerfile.

By using option2 we may also reduce modification of Dockerfile in the future, this may be a good feature because I expect most of our DAGs could be handled by python (and the corresponding libraries) so we don't need to touch installation out of python scope. If we need to install a lot of non-python tools, then I will prefer option1 instead.

Reply:
>  I am not sure what the motivation is to remove poetry (for smaller docker images?)
yes~

> If we want to choose the option2 (use poetry to manage all packages), I would suggest to add more comments in the Dockerfile.
sure, good point~

> By using option2 we may also reduce modification of Dockerfile in the future
agree~

if you're good about option 2 then I'll add more comments in the dockerfile. But also open to option 1~

Reply:
IMO, option 1 might be the better solution. By doing so, we could reduce the risk of dependency conflict between airflow and whatever other packages we gonna use in the future.  As for how `we build the other stuff on top of it.` should be implemented, I think something like `airflow.operators.docker_operator` might be an even better environment isolation mechanism. But I'm good with our current solution.

Reply:
current deployment  is using docker.

I'm not familiar with nested docker but if it's  feasible  then I'm okay with either  haha